packages/sdk/src/shared/react/components/FormattedCurrency.tsx

```diff
@@ -0,0 +1,171 @@
+"use client";
+
+import { B3_COIN_IMAGE_URL } from "@b3dotfun/sdk/shared/constants/currency";
+import { cn } from "@b3dotfun/sdk/shared/utils";
+import { Tooltip, TooltipContent, TooltipTrigger } from "../../../global-account/react/components/ui/tooltip";
+import { useCurrencyConversion } from "../hooks/useCurrencyConversion";
+import { useCurrencyModalStore } from "../stores/currencyModalStore";
+
+interface FormattedCurrencyProps {
+  amount: number;
+  showChange?: boolean;
+  showColor?: boolean;
+  className?: string;
+  subB3Icon?: boolean;
+  clickable?: boolean;
+  decimals?: number;
+  currency?: string; // Override currency (e.g., "ETH", "USDC", "B3")
+}
+
+export function FormattedCurrency({
+  amount,
+  showChange = false,
+  showColor = false,
+  className,
+  subB3Icon = true,
+  clickable = true,
+  decimals,
+  currency,
+}: FormattedCurrencyProps) {
+  const { formatCurrencyValue, selectedCurrency, baseCurrency, exchangeRate } = useCurrencyConversion();
+  const { openModal } = useCurrencyModalStore();
+
+  // Use passed currency or fall back to selected currency
+  const activeCurrency = currency || selectedCurrency;
+
+  const isPositive = amount >= 0;
+
+  // Get the formatted value (using absolute value for negative numbers when showing change)
+  const baseAmount = showChange ? Math.abs(amount) : amount;
+
+  // Use custom decimal formatting and currency override
+  let formattedValue: string;
+  if (currency) {
+    // Custom currency provided - format with decimals if specified
+    if (decimals !== undefined) {
+      const fixed = baseAmount.toFixed(decimals);
+      formattedValue = `${parseFloat(fixed).toString()} ${currency}`;
+    } else {
+      // Use default decimals for the currency
+      const defaultDecimals = currency === "B3" ? 0 : 2;
+      const fixed = baseAmount.toFixed(defaultDecimals);
+      formattedValue = `${parseFloat(fixed).toString()} ${currency}`;
+    }
+  } else if (decimals !== undefined && activeCurrency === "B3") {
+    // Format with specified decimals for B3, then remove trailing zeros
+    const fixed = baseAmount.toFixed(decimals);
+    formattedValue = `${parseFloat(fixed).toString()} B3`;
+  } else {
+    // Use default formatting from hook for all other currencies (and B3 without custom decimals)
+    formattedValue = formatCurrencyValue(baseAmount);
+  }
+
+  // Calculate tooltip value with dynamic decimal handling
+  const getTooltipValue = () => {
```

Comment:
Look like this is same code in packages/sdk/src/shared/react/hooks/useCurrencyConversion.ts?